lib/codegen/src/regalloc/stackmaps.rs

```diff
@@ -0,0 +1,99 @@
+use ir::Function;
+use isa::TargetIsa;
+use dominator_tree::DominatorTree;
+use regalloc::live_value_tracker::LiveValueTracker;
+use regalloc::liveness::Liveness;
+use cursor::{Cursor, FuncCursor};
+use std::collections::HashSet;
+use ir::InstBuilder;
+
+// The emit_stackmaps() function analyzes each instruction to retrieve the liveness of
+// the defs and operands by traversing the dominator tree in a post order fashion.
+pub fn emit_stackmaps(isa: &TargetIsa, func: &mut Function, domtree: &mut DominatorTree,
+    liveness: &mut Liveness, tracker: &mut LiveValueTracker) {
+
+    // Visit EBBs in post-order
+    let mut pos = FuncCursor::new(func);
+    let mut ebbs_for_stackmap = HashSet::new();
+
+    for &ebb in domtree.cfg_postorder().iter() {
+
+        // call ebb_top && drop_dead_params
+        tracker.ebb_top(ebb, &pos.func.dfg, liveness, &pos.func.layout, domtree);
+        tracker.drop_dead_params();
+
+        // From the top of the ebb, step through the instructions
+        pos.goto_top(ebb);
+
+        while let Some(inst) = pos.next_inst() {
+            // Get opcode of instruction
+            let opcode = pos.func.dfg[inst].opcode();
+
+            println!("Instruction Data: {}", pos.func.dfg.display_inst(inst, None));
+
+            // Check if it's a branch instruction
+            if opcode.is_branch() {
+                // Find what the branch destination is
+                let branch_dest = pos.func.dfg[inst].branch_destination();
+                ebbs_for_stackmap.insert(branch_dest);
+            }
+
+            // Check if it's a call instruction
+            if opcode.is_call() {
+                // insert stackmap instruction
+                let mut value_list = Vec::new();
```

Comment:
We should move the `live_value_list` computation below up here, so that we can add the live values to the actual stackmap instruction.